tests/virtual_machines_test.go

```diff
@@ -30,9 +32,34 @@ var _ = Describe("Virtual Machines", func() {
 			Body([]byte(fmt.Sprintf(linuxBridgeConfCRD, "linux-bridge", TestNamespace))).
 			Do()
 		Expect(result.Error()).NotTo(HaveOccurred())
+
+		By("Verify that there are no VMs in the cluster before starting test suite")
+		beforeAllVMList := &kubevirtv1.VirtualMachineList{}
+		err := testClient.VirtClient.List(context.TODO(), beforeAllVMList, &client.ListOptions{})
+		Expect(err).ToNot(HaveOccurred(), "Should successfully list add VMs")
+		Expect(len(beforeAllVMList.Items)).To(BeZero(), "There should be no VM's in the cluster before running test suite")
 	})
 
 	Context("Check the client", func() {
+		BeforeEach(func() {
+			By("Verify that there are no VMs left from previous tests")
```

Comment:
All this clearing can be done in BeforeEach of the whole Describe since all the tests here are dealing with VMs. And if you keep this first check in `Describe` BeforeEach, you can drop the one from BeforeAll

Reply:
Like in this [1] commit?

[1] https://github.com/k8snetworkplumbingwg/kubemacpool/blob/945b21482f692e77f12ffa4fa9d5dd9ad46b3d1b/tests/virtual_machines_test.go

Reply:
Yeah